src/client/datascience/variablesView/variableView.ts

```diff
@@ -138,8 +138,19 @@ export class VariableView extends WebviewViewHost<IVariableViewPanelMapping> imp
         }
         context.set(visible).ignoreErrors();
 
-        // Also perform a refresh on a visibily change
-        this.postMessage(InteractiveWindowMessages.ForceVariableRefresh).ignoreErrors();
+        // I've we've been made visible, make sure that we are updated
+        if (visible) {
```

Comment:
This was a change that we needed along with not retaining context anymore. When we become visible, check if there is a currently active notebook and update the execution count in the variable view. Basically the same operation as if you switch active documents.